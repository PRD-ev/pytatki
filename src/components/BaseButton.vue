<template>
  <button :class="buttonClass.join(' ')">
    <slot></slot>
  </button>
</template>

<script>
import Vue from 'vue';

export default Vue.extend({
  name: 'BaseButton',
  props: {
    size: {
      type: String,
      validator: size => size === 'huge',
    },
  },
  data() {
    return {
      buttonClass: ['button'],
    };
  },
  created() {
    if (this.size === 'huge') {
      this.buttonClass.push('button--huge');
    }
  },
});
</script>

<style scoped lang="scss">
.button {
  font-size: 18px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 10px;
  padding: 7px 15px;
  margin: 10px;
  width: 100%;
  color: var(--white);
  cursor: pointer;
  border: none;
  background: linear-gradient(90deg, var(--orange) 0%, var(--light-orange) 100%);
  &--huge {
    padding: 10px 25px;
    height: 100px;
    width: 250px;
  }
}
</style>
