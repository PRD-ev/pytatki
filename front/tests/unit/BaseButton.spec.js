import { shallowMount } from '@vue/test-utils';
import BaseButton from '@/components/BaseButton.vue';

describe('BaseButton.vue', () => {
  it('Assigns allowed modification class', () => {
    const wrapper = shallowMount(BaseButton, {
      propsData: { size: 'huge' },
    });
    expect(wrapper.find('.button.button--huge').exists()).toBe(true);
  });

  it('Throws error on forbidden modification class', () => {
    const wrapper = shallowMount(BaseButton);
    expect(wrapper.vm.$options.props.size.validator('enormous')).toBe(false);
  });

  it('Matches snapshot (size: huge)', () => {
    const wrapper = shallowMount(BaseButton, {
      propsData: { size: 'huge' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
