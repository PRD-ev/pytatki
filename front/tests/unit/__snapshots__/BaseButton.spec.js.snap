// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton.vue Matches snapshot (size: huge) 1`] = `<button class="button button--huge"></button>`;
